fe/fe-core/src/main/java/org/apache/doris/datasource/metacache/MetaCache.java

```diff
@@ -0,0 +1,111 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.metacache;
+
+import org.apache.doris.common.CacheFactory;
+import org.apache.doris.common.util.Util;
+
+import com.github.benmanes.caffeine.cache.CacheLoader;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.concurrent.ExecutorService;
+
+public class MetaCache<T> {
+    private LoadingCache<String, List<String>> namesCache;
+    private Map<Long, String> idToName = Maps.newConcurrentMap();
+    private LoadingCache<String, Optional<T>> metaObjCache;
+
+    private String name;
+
+    public MetaCache(String name,
+            ExecutorService executor,
+            OptionalLong expireAfterWriteSec,
+            OptionalLong refreshAfterWriteSec,
+            long maxSize,
+            CacheLoader<String, List<String>> namesCacheLoader,
+            CacheLoader<String, Optional<T>> metaObjCacheLoader,
+            RemovalListener<String, Optional<T>> removalListener) {
+        this.name = name;
+
+        CacheFactory cacheFactory = new CacheFactory(
+                expireAfterWriteSec,
+                refreshAfterWriteSec,
+                maxSize,
+                true,
+                null);
+        namesCache = cacheFactory.buildCache(namesCacheLoader, null, executor);
+        metaObjCache = cacheFactory.buildCache(metaObjCacheLoader, removalListener, executor);
+    }
+
+    public List<String> listNames() {
+        return namesCache.get("");
```

Comment:
What does double quotation mark represent

Reply:
Nothing, just use this empty key as key.
Because for `namesCache`, there is one and only one key.